sematic/api/endpoints/resolutions.py

```diff
@@ -515,3 +517,35 @@ def clean_orphaned_resolution_jobs_endpoint(
             content=[change.value for change in state_changes],
         )
     )
+
+
+@sematic_api.route("/api/v1/resolutions/<root_id>/clean", methods=["POST"])
+@authenticate
+def clean_stale_resolution_endpoint(
+    user: Optional[User], root_id: str
+) -> flask.Response:
+    resolution = get_resolution(root_id)
+    root_run = get_run(root_id)
+
+    if not FutureState[root_run.future_state].is_terminal():  # type: ignore
+        return jsonify_error(
+            f"Couldn't clean resolution {root_id} because its root run "
+            f"is in state {root_run.future_state}",
+            status=HTTPStatus.BAD_REQUEST,
```

Comment:
This isn't really a bad request. How about [`409 Conflict`](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#4xx_client_errors)?

Reply:
done